share/spack/templates/container/centos_7.dockerfile

```diff
@@ -0,0 +1,29 @@
+{% extends "container/bootstrap-base.dockerfile" %}
+{% block install_os_packages %}
+RUN yum update -y \
+ && yum install -y epel-release \
+ && yum update -y \
+ && yum --enablerepo epel groupinstall -y "Development Tools" \
+ && yum --enablerepo epel install -y \
+        curl \
+        findutils \
+        gcc-c++ \
+        gcc \
+        gcc-gfortran \
+        git \
+        gnupg2 \
+        hostname \
+        iproute \
+        Lmod \
+        make \
+        patch \
+        python \
+        python-pip \
+        python-setuptools \
+        tcl \
```

Comment:
is this or Lmod directly needed?  Are people using modules w/spack containerize?  Could these ones just be optional?  I guess this is only for the bootstrap image but still.

Reply:
Here I replicated what we currently do for our images on Dockerhub. The idea is that in a following PR we can remove the Dockerfile under `share/spack/docker` and generate them with containerize. In any case I agree that modules are not strictly needed for Spack to work correctly, so I'll try to trim the list of packages here.

Reply:
See ab4533c